### DISCUSSION TITLE
Is there a way to getQueryData() with one key only?

### DISCUSSION THREAD

**Discussion by frederikhors** [TanStack/query#2336] [Q&A]
I'm using the code from the docs: https://sveltequery.vercel.app/guides/placeholder-query-data#placeholder-data-from-cache:

```js
const result = useQuery(["blogPost", blogPostId], () => fetch("/blogPosts"), {
  placeholderData: () => {
    // Use the smaller/preview version of the blogPost from the 'blogPosts' query as the placeholder data for this blogPost query
    return queryClient
      .getQueryData("blogPosts")
      ?.find((d) => d.id === blogPostId);
  },
});
```

My problem is I'm caching `blogPosts` along with pagination info, something like:

```js
let pagination = { page: 1, limit: 20 };

$: posts = useQuery(
  ["blogPosts", pagination],
  async () => await fetchPosts({ pagination })
);
```

Is there a way I can use `placeholderData` with `.getQueryData("blogPosts")` key only (without `pagination`)?

**Comment by TkDodo** [comment]
have a look at `queryClient.getQueryCache().findAll()`, where you can pass `filters` to do fuzzy matching.

**Reply by frederikhors** [reply]
Oh wow. Please can you show an example?

**Reply by TkDodo** [reply]
the possible filters are listed here: https://react-query.tanstack.com/guides/filters

**Reply by frederikhors** [reply]
I tried `.getQueryData("blogPosts")` with `{ exact: false }` and it seems to work!

**Reply by TkDodo** [reply]
ooh yeah, that one, too. But be aware that it gives you the first thing it finds in the cache that matches, so I think it depends on what was added to the cache first. If that's alright for your use-case then 👍

**Reply by frederikhors** [reply]
You were right. It doesn't work for list/detail with pagination.

The first query matched may not contain the post in the e.g. second page.

So I ended up with the below code, what do you think?

```js
const result = useQuery(["blogPost", blogPostId], () => fetch("/blogPosts"), {
  placeholderData: () => {
    let postFound;

    useQueryClient()
      .getQueryCache()
      .findAll("blogPosts", { exact: false })
      .find((query) => {
        return query.state.data?.posts.find((p) => {
          if (p.id === postID) {
            postFound = p;
          }
        });
      });

    return postFound;
  },
});
```

Can you suggest something more elegant?

**Reply by TkDodo** [reply]
looks good, I think you can leave out `exact: false` because for `findAll`, it's the default. Also, you can't call `useQueryClient` in the placeholder callback (violates the rules of hooks), but just make your own custom hook:

```
const useBlogPostQuery = () => {
  const queryClient = useQueryClient()
  return useQuery(...)
}
```

**Reply by frederikhors** [reply]
Oh. Thanks for the `exact`. For the hooks, I'm using Svelte-Query, not React one. Is this different, right?

I'm using `useQueryClient()` documented here: https://sveltequery.vercel.app/reference/useQueryClient.

**Reply by frederikhors** [reply]
In Svelte we don't have hooks (a-la-React).

**Reply by TkDodo** [reply]
oh okay, I have no idea how hook work in svelte 😅 🤷
